### ISSUE TITLE
Change default TLD to stop conflicting with the real internet

### ISSUE THREAD

**Issue by indirect** [boxen/puppet-dnsmasq#16]
So .dev has been sold by ICANN, and has [a registry and root name servers](https://www.iana.org/domains/root/db/dev.html). This means that Boxen is now (potentially) blocking legitimate sites by using .dev for local development. Are there plans to update the default domain to something that isn't actually part of the real internet?

**Comment by celkins** [comment]
Previously: https://github.com/boxen/puppet-dnsmasq/issues/4#issuecomment-55502419

**Comment by spacepants** [comment]
FYI: An update to [Chrome forces .dev domains to HTTPS via preloaded HSTS](https://ma.ttias.be/chrome-force-dev-domains-https-via-preloaded-hsts/) which would effectively break the default TLD for local dev without SSL.

**Comment by jacobbednarz** [comment]
#17 allows the ability for this to be configured for now. I cut the [2.0.3](https://github.com/boxen/puppet-dnsmasq/releases/tag/2.0.3) release a few days ago and before changing the default, I'm allowing cut  over time for everyone to gradually roll it out to their teams (since chrome is the only browser doing this for now).

**Comment by EugenMayer** [comment]
It does not work for me, switched to `.local`, seen the changes to the config and also see them in `/opt/boxen/config/dnsmasq/dnsmasq.conf` ... but still .dev resolves to 127.0.0.1 - .local ist not able to be resolved at all

**Comment by EugenMayer** [comment]
did some digging

so that is there, removed /etc/resolver/dev and also dev.dnsmasq.plist -0 ensured all launchd of dnsmasq are down, and the used this to start it again

`sudo launchctl load /Library/LaunchDaemons/local.dnsmasq.plist  `

while `dig @127.0.0.1 d.local` works and resolves to 127.0.0.1

`dig d.local` will not work. The scoped DNS setting seems to be there 
```
resolver #8
  domain   : local
  nameserver[0] : 127.0.0.1
  flags    : Request A records, Request AAAA records
  reach    : Reachable, Local Address, Directly Reachable Address
 ```

Any clues?

**Comment by EugenMayer** [comment]
Interesting, found

```
resolver #2
  domain   : local
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : Not Reachable
  order    : 300000
```

could that be the cause?

**Comment by EugenMayer** [comment]
Well, that is an issue and expected https://support.apple.com/en-us/HT203136 .. not sure it was a good idea to go for .local here.

**Comment by jacobbednarz** [comment]
> Well, that is an issue and expected https://support.apple.com/en-us/HT203136 .. not sure it was a good idea to go for .local here.

We've moved to `test` internally for that exact reason.

> also one needs to remove /etc/resolver/dev and /etc/resolver/local to cleanup

This was a bug I introduced in #17 however was fixed in #24 a few days back and a new release cut for it. Are you able to update to `2.0.5` and let me know if this fixes the issue for you?
